src/legacy-appcache-behavior-import.js

```diff
@@ -0,0 +1,422 @@
+    return idb.getAll(constants.STORES.MANIFEST_URL_TO_CONTENTS).then(manifests => {
+      log('All manifests:', manifests);
+      // Use .map() to create an array of the longest matching prefix
+      // for each manifest. If no prefixes match for a given manifest,
+      // the value will be ''.
+      const longestForEach = manifests.map(manifestVersions => {
+        // Use the latest version of a given manifest.
+        const parsedManifest =
+          manifestVersions[manifestVersions.length - 1].parsed;
+        return longestMatchingPrefix(
+          Object.keys(parsedManifest.fallback), event.request.url);
+      });
+      log('longestForEach:', longestForEach);
+
+      // Next, find which of the longest matching prefixes from each
+      // manifest is the longest overall. Return both the index of the
+      // manifest in which that match appears and the prefix itself.
+      const longest = longestForEach.reduce((soFar, current, i) => {
+        if (current.length >= soFar.prefix.length) {
+          return {prefix: current, index: i};
+        }
+
+        return soFar;
+      }, {prefix: '', index: 0});
+      log('longest:', longest);
+
+      // Now that we know the longest overall prefix, we'll use that
+      // to lookup the fallback URL value in the winning manifest.
+      const fallbackKey = longest.prefix;
+      log('fallbackKey:', fallbackKey);
+      if (fallbackKey) {
+        const winningManifest = manifests[longest.index];
+        log('winningManifest:', winningManifest);
+        const winningManifestVersion =
+          winningManifest[winningManifest.length - 1];
+        log('winningManifestVersion:', winningManifestVersion);
+        const hash =
+          winningManifest[winningManifest.length - 1].hash;
+        const parsedManifest =
+          winningManifest[winningManifest.length - 1].parsed;
+        return fetchWithFallback(event.request,
+          parsedManifest.fallback[fallbackKey], hash);
+      }
+
+      // If nothing matches, then just fetch().
+      log('Nothing at all matches. Using fetch()');
+      return fetch(event.request);
+    });
+  }
+
+  /**
+   * An attempt to mimic AppCache behavior, using the primitives available to
+   * a service worker.
+   *
+   * @param {FetchEvent} event
+   * @returns {Promise.<Response>}
+   */
+  function appCacheBehaviorForEvent(event) {
+    const requestUrl = new URL(event.request.url);
+    log('Starting appCacheBehaviorForUrl for ' + requestUrl);
+
+    // If this is a request that, as per the AppCache spec, should be handled
+    // via a direct fetch(), then do that and bail early.
+    if (event.request.headers.get('X-Use-Fetch') === 'true') {
+      log('Using fetch() because X-Use-Fetch: true');
+      return fetch(event.request);
+    }
+
+    // Appcache rules only apply to GETs & same-scheme requests.
+    if (event.request.method !== 'GET' ||
+        requestURL.scheme != location.scheme) {
```

Comment:
I think you meant `requestUrl` (honor camelCase) and `protocol` instead of `scheme` (as `scheme` does not exists). The linter is complaining here due to the `!=` as well.